substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ReflectionConfigurationParser.java

```diff
@@ -145,24 +153,9 @@ private void parseFields(List<Object> fields, T clazz) {
     }
 
     private void parseField(Map<String, Object> data, T clazz) {
-        String fieldName = null;
-        boolean allowWrite = false;
-        for (Map.Entry<String, Object> entry : data.entrySet()) {
-            String propertyName = entry.getKey();
-            if (propertyName.equals("name")) {
-                fieldName = asString(entry.getValue(), "name");
-            } else if (propertyName.equals("allowWrite")) {
-                allowWrite = asBoolean(entry.getValue(), "allowWrite");
-            } else if (propertyName.equals("allowUnsafeAccess")) {
-                // ignored (no longer required)
-            } else {
-                throw new JSONParserException("Unknown attribute '" + propertyName + "' (supported attributes: 'name') in definition of field for class '" + delegate.getTypeName(clazz) + "'");
-            }
-        }
-
-        if (fieldName == null) {
-            throw new JSONParserException("Missing attribute 'name' in definition of field for class " + delegate.getTypeName(clazz));
-        }
+        checkAttributes(data, "reflection field descriptor object", Collections.singleton("name"), Collections.singleton("allowWrite"));
```

Comment:
We should still support `allowUnsafeAccess` as a known, but no longer used, attribute that doesn't trigger failure in strict mode.

Reply:
This is a good point - I've returned the `allowUnsafeAccess` attribute in the optional attributes.